### PR TITLE
Adding clippy CI (implementing #24)

### PR DESCRIPTION
**Pull request by jzbor**

### CODE REVIEW

**Comment by GamerBene19** [comment]
https://github.com/mensatt/image-service/blob/main/CONTRIBUTING.md
Thx

**Comment by stormofice** [comment]
Resolves #24.

**Review by StealWonders — APPROVED** [review]

File: src/main.rs [hunk 1]
```diff
@@ -1,3 +1,6 @@
+#![allow(dead_code)]
+#![allow(clippy::redundant_field_names)]
```
Comment (GamerBene19) [inline src/main.rs:2]: Imo it's cleaner to fix the `height: height` issue instead of disabling the check globally

File: src/main.rs [hunk 2]
```diff
@@ -1,3 +1,6 @@
+#![allow(dead_code)]
```
Comment (GamerBene19) [inline src/main.rs:1]: Is there a way to explicitly place this before dead code?

**Review by GamerBene19 — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 1 above)
Comment (stormofice) [reply src/main.rs:2]: I do not find it very consistent to specify some but not all fields, even if clippy prefers this.

File: src/main.rs (hunk 2 above)
Comment (stormofice) [reply src/main.rs:1]: Sure

File: src/main.rs (hunk 1 above)
Comment (GamerBene19) [reply src/main.rs:1]: Agreed

**Review by GamerBene19 — APPROVED** [review]

**Pull request merged**